### ISSUE TITLE
Object disappearing in certain angle in old bve route.

### ISSUE THREAD

**Issue by adfriz** [leezer3/OpenBVE#907]
### Description

as title said

### Reproduction

just play the route, jump to last station, go to track view (f3), jump to last station in track view, look at left building.

## Route

Hakone tozan railway. 
https://web.archive.org/web/20130210120612/http://www5.pf-x.net/~twingle/bve_hakone.html

note: you need to put the route into standard route structure. 
 Railway 
      | - Object (the object data zip)
      | - Route (route data zip)
      | - Sound (sound data 1 and 2 zip, need to put all the sound in this Sound folder)
## Train

Type 2000 St. Moritz (2-car train)
https://web.archive.org/web/20130210120612/http://www5.pf-x.net/~twingle/bve_train.html

## Logs

no crash, no logs.

### Related information 

* Operating system: windows 10
* openbve 1.9.1.1 64 bit installer version.
* Method of control : Keyboard and mouse

https://github.com/leezer3/OpenBVE/assets/76892624/f0989a1a-a068-49fb-a626-de932cb2c4e7

**Comment by leezer3** [comment]
Will take a proper look later in the week, but that's *probably* normal / expected.
TLDR: From the video, I'd suspect its using Options.ObjectVisibility set to 0 (or not set at all)
When an object with a negative length is in view, rotating the camera to move it's source block out of the field of view will cause this effect.
It's pretty common on BVE2 era content, although not always as noticeable as this.
If so, may be able to add a routefile hack to workaround this, but it's not a bug.

**Comment by leezer3** [comment]
Having tested, it's definitely the ObjectVisibility setting causing this. 

However, this is one route I'm not happy changing the setting on internally, and this is unfortunately one of those things.... 
Doing this causes a different set of glitches with the change ends feature, and various bits of ground appearing through the track. (Caused as it's a massive set of switchbacks)
